### ISSUE TITLE
Compound index can't contain autoincrementing primary key

### ISSUE THREAD

**Issue by CiriousJoker** [dfahlander/Dexie.js#751]
Title says it all pretty much.

`
                ++id,
                questionId,
                timestamp,
                action,
                executing,
                questionlistId,
                [questionId+executing+questionlistId]`
I can query for

**Comment by dfahlander** [comment]
I recall this has been reported before. There should be another dexie issue about it. The issue seems to be on the indexedDB layer in the browser. Don't recall whether it was Chrome, Safari or Firefox or several of them. If you find the other issue, please link it to this.

**Comment by dfahlander** [comment]
This has been reported in #648 as well. It's most certainly a browser bug. I have an old todo-list there to figure out which browsers are affected. In #648, the reporter states that the bug is there in both Chrome and Firefox. Those two implementations are otherwise the most bug-free and they react quite quickly on bug reports, so I suppose bug reports in their repos will be the thing to do. But first verify what browsers are affected.

**Comment by dfahlander** [comment]
Only Safari behaves well in this case.

| Browser | Version | Status |
| -------- | --- |:------- |
| *IE* | 11 | *No compound index support* |
| *Edge* | Evergreen / Oct 2018 | *No compound index support* |
| Chrome | Evergreen / Oct 2018 | Fails to set compound index when part of autoIncremented |
| Firefox | Evergreen / Oct 2018 | Fails to set compound index when part of autoIncremented |
| **Safari** | Evergreen / Oct 2018 | **Behaves as expected** |

(Nice work, @beidson )

**Comment by sechel** [comment]
I use a workaround in this situation and store the entry in question right after the `put` call again including the auto generated key. Then all compound indices are working as expected. Are there corresponding issues at Mozilla and Google?

**Comment by sechel** [comment]
Here is a plunk that demonstrates this: https://run.plnkr.co/preview/cjuy3pv3l00063h5wdo9t2vlj/

**Comment by sechel** [comment]
Chrome seems to have fixed this in the meantime. Firefox remains broken.

**Comment by aggregat4** [comment]
I'm on Chrome 77.0.3865.35  (beta) and I still see this.

This JSFiddle does not return results for me on Chrome: https://jsfiddle.net/xonm9fe0/17/

Does the "between" query make a difference?

**Comment by sechel** [comment]
You are right, I concluded too fast. It fails in the corresponding unit test for me, too. The thing is that didn't have the unit test and workaround in place before. And it did work in my other tests (and in production) so there may an easier workaround than just `putting` it twice (which is the workaround that I'm using now). But I haven't had the time so sort this out by now.

**Comment by sechel** [comment]
Since Chrome `78` this has been fixed now.

**Comment by sechel** [comment]
Firefox `129` (currently on the Nightly build) seems to have fixed this too. 🥳

**Comment by dfahlander** [comment]
> At least since Firefox `126` this seems to have been fixed too. 🥳

I just verified that it's fixed in Firefox 126 ([released May 14, 2024](https://www.mozilla.org/en-US/firefox/126.0/releasenotes/)) but that the bug was still there in Firefox 125.
